client/src/app/recipe-list/recipe-list.component.html

```diff
@@ -1,19 +1,35 @@
-<ng-container *ngIf="recipeList.length != 0">
-  <ng-container *ngIf="recipeList.length == 1; else moreThanOne">
-    <h3>There is {{ recipeList.length }} recipe in the list.</h3>
-    <div *ngFor="let recipe of recipeList">
-      <a [routerLink]="['/', appPaths.recipe, recipe.id]">{{ recipe.name }}</a>
-    </div>
+<ng-container *ngIf="isLoading; else content">
+  <p>Loading...</p>
+</ng-container>
+<ng-template #content>
+  <ng-container *ngIf="recipeList.length > 0; else zeroRecipes">
+    <ng-container *ngIf="recipeList.length == 1; else moreThanOne">
+      <h3>There is {{ recipeList.length }} recipe in the list.</h3>
+      <div *ngFor="let recipe of recipeList">
+        <a [routerLink]="['/', appPaths.recipe, recipe.id]">{{
+          recipe.name
+        }}</a>
+      </div>
+    </ng-container>
+    <ng-template #moreThanOne>
+      <h3>
+        There are {{ recipeList.length }} recipes in the list that contain the
+        term '{{ searchedQuery }}'.
+      </h3>
+      <div *ngFor="let recipe of recipeList">
+        <a [routerLink]="['/', appPaths.recipe, recipe.id]">{{
+          recipe.name
+        }}</a>
+      </div>
+    </ng-template>
   </ng-container>
-  <ng-template #moreThanOne>
-    <h3>
-      There are {{ recipeList.length }} recipes in the list that contain the term
-      '{{ searchedQuery }}'
-    </h3>
-    <div *ngFor="let recipe of recipeList">
-      <a [routerLink]="['/', appPaths.recipe, recipe.id]">{{ recipe.name }}</a>
-    </div>
+  <ng-template #zeroRecipes>
```

Comment:
This still needs to be handled better, having multiple templates here for almost the same thing isn't efficient. Consider using ngIf without the use of templates.